client/my-sites/post-share/connections-list.jsx

```diff
@@ -54,10 +54,15 @@ class ConnectionsList extends PureComponent {
 			return null;
 		}
 
+		const brokenConnections = connections.filter( connection => connection.status === 'broken' );
+
+		if ( ! brokenConnections || ! brokenConnections.length ) {
```

Comment:
Shouldn't have to check if `brokenConnections` exists at this point since we know `connections.filter`  will return an array.

You only need to check the length:
```
if (  ! brokenConnections.length ) {
   return null
}
```